a653rs-router/src/ports.rs

```diff
@@ -36,11 +36,14 @@ impl<S: ApexSamplingPortP4> RouterOutput for SamplingPortSource<S> {
 
 impl<Q: ApexQueuingPortP4> RouterInput for QueuingPortReceiver<Q> {
     fn receive<'a>(&self, buf: &'a mut [u8]) -> Result<&'a [u8], PortError> {
-        let timeout = SystemTime::Normal(Duration::from_micros(10));
+        let timeout = SystemTime::Normal(Duration::from_secs(0));
```

Comment:
There is a constant for zero.  
https://doc.rust-lang.org/std/time/struct.Duration.html#associatedconstant.ZERO